SatelliteQE/robottelo
# Dynaconf CI changes including native configuration for Airgun and Nailgun

This PR includes:
--------------------------
0. Merged PR #8246 in this PR. which contains:
    - Additions of DynaConf YAML templates for 2 sections- Robottelo and Server. 
    - Inclusion of missing repos in Robottelo from Gitlab and 
    - Updates to tests wrt repo changes
1. Configuration for Nailgun and Airgun moved SettingsNodeWrapper which fetches settings from Dynaconf / Legacy settings using settings Facade.
2. Calling Nailgun and Airgun settings funcs post settings are set from facade (Obiviously) :)
3. Removed Nailgun and Airgun configuration from legacy settings object, due to (1).
4. New Robottelo dispatch value func which returns data using settings object for DynaConf as `setings.screenshots_path` instead of `settings.robottelo.screenshots_path`. This notation is being used by all the tests in robottelo, so this saves from big refactoring.
5. Separate repos.yaml and swapped the places of existing settings in robottelo and server YAMLs.
6. Logging updated for facade based settings object. robottelo and server settings base config removed.

Backporting Commit:
-----------------------------
The back-porting commit https://github.com/SatelliteQE/robottelo/commit/3fdb58ff63f68500d7f106d71695d31502b7c747 is added to make PR / facade compatible with existing / old pipeline. The commit should be removed as soon as we move to new pipeline. See below.

TO-DOs for future (OR when we start using new Pipeline):
-------------------------------------------------------------------------------
1. Uncommit the commit https://github.com/SatelliteQE/robottelo/commit/3fdb58ff63f68500d7f106d71695d31502b7c747 and rebase with that times Master.

TO-DOs for future (OR when we translate all robottelo.settings sections to DynaConf settings):
--------------------------------------------------------------------------------------------------------------------------------
1. Remove facade and all its methods.
2. Remove all the methods from `robottelo.config.base` of legacy settings.
3. Update Unit-tests for removals of legacy settings and additions of dynaconf settings.
